src/scene/forward-renderer.js

```diff
@@ -974,11 +974,17 @@ pc.extend(pc, function () {
             var i, drawCall, visible;
             var drawCallsCount = drawCalls.length;
 
+            var cullingMask = camera.cullingMask || 0xFFFFFFFF; // if missing assume camera's default value
+
             if (!camera.frustumCulling) {
                 for (i = 0; i < drawCallsCount; i++) {
                     // need to copy array anyway because sorting will happen and it'll break original draw call order assumption
                     drawCall = drawCalls[i];
                     if (!drawCall.visible && !drawCall.command) continue;
+
+                    // if the object's mask AND the camera's cullingMask is zero then the game object will be invisible from the camera
+                    if (drawCall.mask && ((1 << drawCall.mask) & cullingMask) === 0) continue;

```

Comment:
You don't actually need "1 <<" here, because drawCall.mask is already a bitmask ready to be AND'ed